scripts/test-all.sh

```diff
@@ -33,25 +33,80 @@ run_test() {
     echo -e "${BLUE}Running $name...${NC}"
 
     if (cd "$(dirname "$0")/.." && eval "$command") > /tmp/test_output 2>&1; then
-        echo -e "${GREEN}✓ PASS${NC} - $name"
-        RESULTS+=("PASS: $name")
+        # Check if tests were skipped
+        skip_count=$(grep -c "^--- SKIP:" /tmp/test_output 2>/dev/null || true)
+        skip_count=${skip_count:-0}
+        pass_count=$(grep -c "^--- PASS:" /tmp/test_output 2>/dev/null || true)
+        pass_count=${pass_count:-0}
+        
+        if [ "$skip_count" -gt 0 ] && [ "$pass_count" -eq 0 ]; then
+            # All tests were skipped
+            echo -e "${YELLOW}⚠ SKIP${NC} - $name (all tests skipped)"
+            RESULTS+=("SKIP: $name")
+            
+            # Show skip reasons when verbose
+            if [ "$VERBOSE" = true ]; then
+                echo -e "${YELLOW}Skipped tests:${NC}"
+                grep -A1 "^--- SKIP:" /tmp/test_output | grep -v "^--$" | sed 's/^/  /' | while read line; do
```

Comment:
Consider adding a minor improvement: The parsing logic could be made more robust by using `grep -E` with word boundaries to ensure exact matches: `grep -Ec "^--- SKIP:\s"`. This would prevent false positives if test output contains similar patterns in logged messages.